dataformat-xml-dom/src/test/java/org/camunda/spin/impl/xml/dom/format/DomXmlDataFormatWriterTest.java

```diff
@@ -0,0 +1,118 @@
+package org.camunda.spin.impl.xml.dom.format;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.StringWriter;
+import javax.xml.transform.Templates;
+import org.camunda.spin.impl.util.SpinIoUtil;
+import org.camunda.spin.xml.SpinXmlElement;
+import org.junit.Test;
+import org.w3c.dom.Element;
+
+/**
+ * Test xml transformation in DomXmlDataFormatWriter
+ */
+public class DomXmlDataFormatWriterTest {
```

Comment:
* I have trouble understanding some of the logic of the tests in this class: for example why the "clones" are needed. 
* I don't think we need to test the lazy initialization as it's an implementation detail (that might change in the future). We should rather cover it with other tests.

Instead, I suggest to write tests with Spin's API, i.e. using `SpinXmlElement` instead of the the reader/writer classes. I wrote a test here that you could reuse: https://github.com/camunda/camunda-spin/commit/700f6e2103f7eee6a0d79f596a249bd5e4f9ca81

Reply:
completely rewritten the test on base of Spin as suggested in your commit. There is still one TODO marker in testDisabledPrettyPrintFormatted because it removes a blank line and I do not know why. On the other side, this is a test for a new feature as today's implementation always formats the XML. So it also does not break anything.

Reply:
ok, removed the TODO